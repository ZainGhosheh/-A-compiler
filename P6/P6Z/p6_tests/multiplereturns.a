main : () -> int {
  if (true) {
    return 1;
  }
  else 
  {
    a:int = 0;
  }

  return 3;
}