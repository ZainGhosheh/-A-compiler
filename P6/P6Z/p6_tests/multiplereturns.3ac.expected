[BEGIN GLOBALS]
main
[END GLOBALS]
[BEGIN <init> LOCALS]
[END <init> LOCALS]
fun_<init>: enter <init>
lbl_0:      leave <init>
[BEGIN main LOCALS]
a (local var of 8 bytes)
[END main LOCALS]
main:       enter main
            IFZ 1 GOTO lbl_2
            setret 1
            goto lbl_1
            goto lbl_3
lbl_2:      nop
            [a] := 0
lbl_3:      nop
            setret 3
            goto lbl_1
lbl_1:      leave main